src/shared/observables/createOptionStream.ts

```diff
@@ -7,14 +7,15 @@ import { Options } from '../options';
 import { vscode } from '../../vscodeAdapter';
 import { Observable, Observer } from 'rxjs';
 import { publishBehavior } from 'rxjs/operators';
+import { csharpDevkitExtensionId } from '../../utils/getCSharpDevKit';
 
 export default function createOptionStream(vscode: vscode): Observable<Options> {
     return Observable.create((observer: Observer<Options>) => {
         const disposable = vscode.workspace.onDidChangeConfiguration((e) => {
             //if the omnisharp or csharp configuration are affected only then read the options
             if (
                 e.affectsConfiguration('dotnet') ||
-                e.affectsConfiguration('omnisharp') ||
+                (e.affectsConfiguration('omnisharp') && !vscode.extensions.getExtension(csharpDevkitExtensionId)) ||
```

Comment:
Should this logic instead move to somewhere like

https://github.com/dotnet/vscode-csharp/blob/f01115f20d428a704df42a55ab8ea31816beb2d8/src/shared/options.ts#L349

It'll probably work either way, but I'm somewhat surprised already that the Omnisharp option isn't being ignored by that.